lib/moonshot/command_line.rb

```diff
@@ -4,12 +4,23 @@ module Moonshot
   # This class implements the command-line `moonshot` tool.
   class CommandLine
     def self.run! # rubocop:disable AbcSize, CyclomaticComplexity, MethodLength, PerceivedComplexity
+      # Handling premature early command like
+      # `moonshot new` or `moonshot --help`
+      catch(:regular_command) do
```

Comment:
I don't have a really compelling reason *why*, but I don't like the use of catch/throw between methods here. Might be simpler if `handle_early_commands` returns true when it has handled an early command and you can simply write:
```ruby
return if handle_early_commands(ARGV[0..1])
```